diff --git a/spotify_player/src/cli/handlers.rs b/spotify_player/src/cli/handlers.rs
index 70dfb66..f180038 100644
--- a/spotify_player/src/cli/handlers.rs
+++ b/spotify_player/src/cli/handlers.rs
@@ -187,7 +187,8 @@ pub fn handle_cli_subcommand(cmd: &str, args: &ArgMatches) -> Result<()> {
     match cmd {
         "authenticate" => {
             let auth_config = AuthConfig::new(configs)?;
-            crate::auth::get_creds(&auth_config, true, false)?;
+            let creds = crate::auth::get_creds(&auth_config, true, false)?;
+            auth_config.cache.save_credentials(&creds);
             std::process::exit(0);
         }
         "generate" => {
